crates/nu-plugin/src/plugin/mod.rs

```diff
@@ -708,6 +428,13 @@ pub fn serve_plugin(plugin: &impl StreamingPlugin, encoder: impl PluginEncoder +
         std::process::exit(1)
     }
 
+    // Build commands map, to make running a command easier
+    let commands: HashMap<String, _> = plugin
+        .commands()
+        .into_iter()
+        .map(|command| (command.signature().sig.name.clone(), command))
```

Comment:
Not really sure: Should we output warning messages if there are two commands have the same name?

Reply:
I thought about that too. I think it's a good idea.